### PR TITLE
Stac 8617

### PR DESCRIPTION
**Pull request by craffit**
### Step 1: Link to Jira issue

STAC-8617

### Step 2: Description of changes

Add molecule tests for container-to-container and pod-to-pod through headless service

### Step 3: Did you add / update tests for your changes in the right area?
- [ ] Unit/Component test
- [ ] Integration test
- [ ] E2E/Molecule test

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Did you add release notes describing the changes you made?
- [ ] Yes

### Step 6: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: test/molecule-role/molecule/kubernetes/tests/test_healthy.py [hunk 1]
```diff
@@ -47,3 +47,23 @@ def test_apply_pod_to_service_demo(host, ansible_var):
     # so we make sure we deploy the demo after the node agent is healthy
     c = kubeconfig_env + "kubectl -n={} apply -f /home/ubuntu/deployment/test_connections/pod-to-service-cluster-ip.yaml".format(namespace)
     assert host.run(c).rc == 0
+
+
+@pytest.mark.fifth
+def test_apply_container_to_container_demo(host, ansible_var):
+    namespace = ansible_var('namespace')
+
+    # We recognize DNAT connections after the agent is started, because we dunno directions of in-flight connections
```
Comment (TizianoPerrucci) [inline test/molecule-role/molecule/kubernetes/tests/test_healthy.py:56]: update comment

File: test/molecule-role/molecule/kubernetes/tests/test_healthy.py [hunk 2]
```diff
@@ -47,3 +47,23 @@ def test_apply_pod_to_service_demo(host, ansible_var):
...
+    # so we make sure we deploy the demo after the node agent is healthy
+    c = kubeconfig_env + "kubectl -n={} apply -f /home/ubuntu/deployment/test_connections/pod-localhost.yaml".format(namespace)
+    assert host.run(c).rc == 0
+
+
+@pytest.mark.sixth
+def test_apply_pod_to_pod_headless_demo(host, ansible_var):
+    namespace = ansible_var('namespace')
+
+    # We recognize DNAT connections after the agent is started, because we dunno directions of in-flight connections
```
Comment (TizianoPerrucci) [inline test/molecule-role/molecule/kubernetes/tests/test_healthy.py:66]: update comment

File: test/molecule-role/molecule/vms/tests/test_receiver_topology.py [hunk 3]
```diff
@@ -108,7 +108,7 @@ def wait_for_components():
         with open("./topic-topo-process-agents-dnat.json", 'w') as f:
             json.dump(json_data, f, indent=4)
 
-        endpoint_match = re.compile("urn:endpoint:/.*:{}".format(ubuntu_private_ip))
+        endpoint_match = re.compile("urn:endpoint:/{}".format(ubuntu_private_ip))
```
Comment (TizianoPerrucci) [inline test/molecule-role/molecule/vms/tests/test_receiver_topology.py:111]: why has this changed?

Comment (craffit) [reply test/molecule-role/molecule/vms/tests/test_receiver_topology.py:111]: due to the fact that we do not produce a namespace when the connection is not on loopback. this was basically triggering the bug that i fixed

**Review by TizianoPerrucci — APPROVED** [review]

**Pull request merged**